src/goto-programs/json_goto_trace.cpp

```diff
@@ -114,6 +106,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          assert(step.full_lhs_value.is_not_nil());
+          full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
+        }
+        else
+        {
+          if(step.full_lhs_value.is_not_nil())
```

Comment:
Looking more in details at one of the examples, most of the symbols are in mode C but there are some `invalid_object` for which the mode is an empty string.
One fix would be to revert to the default language when we fail to get the language.
Another could be to set a mode to invalid objects...